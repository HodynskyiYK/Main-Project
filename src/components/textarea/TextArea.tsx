import React, { FC } from 'react'
import classnames from 'classnames'
import { ITextArea } from './textareaTypes'
import styles from './TextArea.module.scss'

export const TextArea: FC<ITextArea> = (
    {
        textAreaName,
        textAreaPlaceholder,
        labelText,
        textAreaValue,
        onChange,
        onBlur
    }) => {

    return (
        <>
            {
                labelText && (
                    <label
                        className={styles.Label}
                        htmlFor={textAreaName}
                    >{labelText}</label>
                )
            }
            <textarea
                className={classnames(styles.TextArea)}
                name={textAreaName}
                id={textAreaName}
                cols={30}
                rows={7}
                placeholder={textAreaPlaceholder}
                value={textAreaValue}
                onChange={onChange}
                onBlur={onBlur}
            />
        </>
    )
}
