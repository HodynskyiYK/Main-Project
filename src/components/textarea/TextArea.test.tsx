import React from 'react'
import { render, screen } from '@testing-library/react'
import { TextArea } from './TextArea'


describe('TextArea tests', () => {

    test('To match snapshot', () => {
        const {asFragment} = render(<TextArea textAreaName={'test'} />)
        expect(asFragment()).toMatchSnapshot()
    })

    test('label text', () => {
        render(<TextArea textAreaName={'test'}  labelText={'label text'} />)
        const label = screen.getByText(/label text/i)
        expect(label).toBeInTheDocument()
    })
})