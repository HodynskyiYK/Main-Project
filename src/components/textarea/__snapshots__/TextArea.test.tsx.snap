// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextArea tests To match snapshot 1`] = `
<DocumentFragment>
  <textarea
    class="TextArea"
    cols="30"
    id="test"
    name="test"
    rows="7"
  />
</DocumentFragment>
`;
